db/compaction/compaction_outputs.cc

```diff
@@ -297,13 +345,10 @@ Status CompactionOutputs::AddToOutput(
       return s;
     }
     // reset grandparent information
-    const std::vector<FileMetaData*>& grandparents =
-        compaction_->grandparents();
-    grandparent_overlapped_bytes_ =
-        being_grandparent_gap_
-            ? 0
-            : grandparents[grandparent_index_]->fd.GetFileSize();
+    grandparent_overlapped_bytes_ = 0;
     grandparent_boundary_switched_num_ = 0;
+    grandparent_overlapped_bytes_ =
```

Comment:
Nit: This line and line 348 confused me initially: we assign to `grandparent_overlapped_bytes_` consecutively. I see the assertion assert(grandparent_overlapped_bytes_ == 0); is already done in `UpdateGrandparentBoundaryInfo()` before calling `GetCurrentKeyGrandparentOverlappedBytes()`. Maybe we can assign GetCurrentKeyGrandparentOverlappedBytes(key) to grandparent_overlapped_bytes_ directly, after removing the assertion in `GetCurrentKeyGrandparentOverlappedBytes()`?

Reply:
Sure, sounds good.